src/main/java/com/iridium/iridiumenchants/effects/VeinMine.java

```diff
@@ -0,0 +1,201 @@
+                            .replace("%tool%", WordUtils.capitalize(displayName.replace("_", " ")))
+                    ));
+                    return;
+                }
+
+                if(usedTool.getItemMeta() instanceof Damageable) {
+
+                    int toolDamage = damageTool(blockCount, usedTool, chance);
+
+                    if (safetyFired) {
+
+                        String displayName;
+                        if(usedTool.getItemMeta().hasDisplayName()) displayName = usedTool.getItemMeta().getDisplayName();
+                        else displayName = usedTool.getType().toString().toLowerCase();
+
+                        player.sendMessage(StringUtils.color(IridiumEnchants.getInstance().getMessages().cannotVeinMine
+                                .replace("%prefix%", IridiumEnchants.getInstance().getConfiguration().prefix)
+                                .replace("%tool%", WordUtils.capitalize(displayName.replace("_", " ")))
+                        ));
+                        safetyFired = false;
+                        return;
+                    }
+
+                    Damageable toolMeta = (Damageable) usedTool.getItemMeta();
+                    toolMeta.setDamage(toolDamage);
+                    usedTool.setItemMeta(toolMeta);
+                }
+
+                if (IridiumEnchants.getInstance().getConfiguration().veinMinerDropsAtBreakLocation) {
+                    ArrayList<Collection<ItemStack>> blockDrops = new ArrayList<>();
+
+                    for (Block adjacentBlock : blocksToMine) {
+                        blockDrops.add(adjacentBlock.getDrops(usedTool));
+                    }
+
+                    for (Collection<ItemStack> drops : blockDrops) {
+                        for (ItemStack item : drops) {
+                            eventBlock.getLocation().getWorld().dropItemNaturally(eventBlock.getLocation(), item);
+                        }
+                    }
+
+                } else {
+                    for (Block adjacentBlock : blocksToMine) {
+                        for (ItemStack item : adjacentBlock.getDrops(usedTool)) {
+                            adjacentBlock.getLocation().getWorld().dropItemNaturally(adjacentBlock.getLocation(), item);
+                        }
+                    }
+                }
+
+                for(Block block : blocksToMine) {
+                    block.setType(getAir(block));
+                }
+            }
+        }
+    }
+
+    private boolean isTool(ItemStack itemStack) {
+        return itemStack.getType().name().toUpperCase().contains("_AXE")
+                || itemStack.getType().name().toUpperCase().contains("PICKAXE")
+                || itemStack.getType().name().toUpperCase().contains("SHOVEL")
+                || itemStack.getType().name().toUpperCase().contains("HOE");
+    }
+
+    private int damageTool(int blockCount, ItemStack itemStack, int chance) {
```

Comment:
maybe rename this to something like getToolDamage? damageTool sounds like it applys the damage to the tool